lib/addons/fluxcd/index.ts

```diff
@@ -7,51 +7,102 @@ import { createNamespace, supportsALL } from "../../utils";
 import { HelmAddOn, HelmAddOnProps, HelmAddOnUserProps } from "../helm-addon";
 import { FluxGitRepository } from "./gitrepository";
 import { FluxKustomization } from "./kustomization";
+import { FluxBucket } from './bucket';
 import { KubernetesManifest } from 'aws-cdk-lib/aws-eks/lib/k8s-manifest';
 
 /**
  * Options for a FluxCD GitRepository
  * path and name parameter within repository parameter do not have any affect in flux, may have an affect in argo
  */
 export interface FluxGitRepo extends Required<spi.GitOpsApplicationDeployment> {
-  /** 
-  * Flux SecretRef */
-  secretRefName?: string;
-
-  /** 
-  * Internal for Flux sync.
-  * Default `5m0s` */
-  syncInterval?: string;
-
-  /**
-  * List of kustomizations to create from different paths in repo. */
-  kustomizations?: FluxKustomizationProps[];
+    /** 
+     * Flux SecretRef
+     */
+    secretRefName?: string;
+
+    /** 
+     * Internal for Flux sync.
+     * Default `5m0s`
+     */
+    syncInterval?: string;
+
+    /**
+     * List of kustomizations to create from different paths in repo
+     */
+    kustomizations?: FluxKustomizationProps[];
+}
+
+/**

```

Comment:
As far as design - the`repository` construct is effectively misused as a vehicle to carry bucket info. Consider a case when we add a URL validator  (lack of it is a miss) to the repoUrl and start failing valid cases for the bucket name.
why not omit this structure from the type and introduce a clear bucket type?

Reply:
see commit 1c95b13078e8ea6406b7de009220763952bd5e0d

- [ ] test bucket construct changes